app/queries/with_matching_values_for_all_keys.rb

```diff
@@ -1,13 +1,16 @@
-class WithOneMatchingValueForEachKey
+class WithMatchingValuesForAllKeys
```

Comment:
It occurs to me that this class name is probably trying too hard to explain the implementation rather than what this object is for. What if you simply called it something like `MatchedForNotification`, leaving the explanation of the (not exactly simple) query in the comment?